### ISSUE TITLE
Renaming/moving items (Transparent Sync) does not delete original .yml

### ISSUE THREAD

**Issue by Krusen** [kamsar/Unicorn#129]
When renaming or moving items using transparent sync and `useDataCache=true` the original .yml is not deleted from the filesystem. Having `useDataCache=false` works fine.

I've tested with v3.1.4 on Sitecore 8.1 rev. 151207 (Update-1).

This is the configuration I've tested with:

```XML
<configuration name="Default Configuration" description="">
  <dataProviderConfiguration enableTransparentSync="true" type="Unicorn.Data.DataProvider.DefaultUnicornDataProviderConfiguration, Unicorn" />
  <targetDataStore type="Rainbow.Storage.SerializationFileSystemDataStore, Rainbow" physicalRootPath="$(dataFolder)\Unicorn\$(configurationName)" useDataCache="true" singleInstance="true"/>

  <predicate type="Unicorn.Predicates.SerializationPresetPredicate, Unicorn" singleInstance="true">
    <include database="master" path="/sitecore/templates/User Defined" />
  </predicate>
</configuration>
```

Adding a template and then renaming it or moving it to a subfolder causes the issue.

**Comment by Krusen** [comment]
I believe I've found the issue and also maybe a fix.

When moving/renaming an item, the item that gets saved into the cache has the default value (`(from Sitecore database)`) for `SerializedItemId` while it should instead be the file path. After the first move/rename this is what is loaded from the cache and it then tries to delete the file `(from Sitecore database)` instead of the actual file.

Setting `SerializedItemId` to the file path as demonstrated below seems to fix the issue. I'm unsure if it affects anything else, but it looks okay from my limited knowledge of Rainbow.

https://github.com/kamsar/Rainbow/blob/master/src/Rainbow/Storage/SerializationFileSystemTree.cs#L352

Current: `var proxiedItem = new ProxyItem(item);`
Fixed: `var proxiedItem = new ProxyItem(item) {SerializedItemId = path};`

**Comment by kamsar** [comment]
I cherry picked your commits and agree that they fix the issue, which I was able to reproduce.

Thank you for your well researched and reported issue, AND the code to fix it. With a test that failed without the tree change. Love it.

**Comment by kamsar** [comment]
This should be fixed in Unicorn 3.1.5-beta01, available on NuGet.

**Comment by kamsar** [comment]
Unicorn 3.1.5 has been released with this fix. Closing the issue. If you find it again, please reopen.
